hook!-full-game/addons/level_design_dock/LevelDesignDock.gd

```diff
@@ -0,0 +1,27 @@
+tool
+extends Control
+
+signal packed_scene_button_created(button)
+onready var grid_container: GridContainer = $VBoxContainer/GridContainer
+
+const PACKED_SCENE_BUTTON: = preload("res://addons/level_design_dock/PackedSceneButton.tscn")
+const PackedSceneButton: = preload("res://addons/level_design_dock/PackedSceneButton.gd")
+
+func drop_data(position: Vector2, data: Dictionary) -> void:
+	var packed_scene_path: String = data.files[0] as String
+	create_packed_scene_button(packed_scene_path)
+
+
+func can_drop_data(position: Vector2, data: Dictionary) -> bool:
+	var can_drop: = false
+	if data.type == "files":
```

Comment:
```gdscript
if data.type != "files":
    return false
return load(data.files[0]) is PackedScene
```

Although I wouldn't use a load statement. Isn't checking that the file extension is .tscn possible? And is it really okay to drop the data if only the first file in the array is a packed scene?

Instead I'd have a function or loop that filters the files:
```gdscript
valid_files: = []
for path in files:
    if path.endswith('.tscn'):
         valid_files.append(path)
return valid_files

[...]

can_drop = valid_files.size() > 0
```

Reply:
Oh...for sure, I even tried to see if I could access EditorFyleSystem to check the resource type...and forgot completely that PackedScenes are the only (textual) Resources with an exclusive name! :rofl: I will fix it, sorry